### PR TITLE
FEATURE: Support ``.fusion`` file extension for Fusion

### LINKED ISSUE #1267: Neos.Fusion should respect ``.fusion`` files
Since the TYPO3.TypoScript is now renamed to Neos.Fusion the rendering should also respect `.fusion` files. For compatibility reasons `.ts2` files should still be processed.

### PR DESCRIPTION
**Pull request by johannessteu**
This changes introduces the support of ``.fusion`` file extension to parse Fusion. The support for ``.ts2`` file extension is now marked as deprecated and will be dropped with the next major release.

Resolves #1267

### CODE REVIEW

File: Neos.Fusion/Classes/Core/Parser.php [hunk 1]
```diff
@@ -609,13 +609,12 @@ protected function parseInclude($include)
...
             $iterator = new \DirectoryIterator($basePath);
         }
-
         // If iterator is set it means we're doing globbing
         if (isset($iterator)) {
             foreach ($iterator as $fileInfo) {
                 $pathAndFilename = $fileInfo->getPathname();
-                // Only work on .ts2 files
-                if ($fileInfo->getExtension() === 'ts2') {
+                // Only work on .fusion files and .ts2 files. The support for .ts2 is a fallback and will be dropped with 4.0
```
Comment (skurfuerst) [inline Neos.Fusion/Classes/Core/Parser.php]: Please add the word "deprecated" somewhere in the comment, as this is what we usually search for to find these deprecations in code :)

File: Neos.Fusion/Tests/Unit/Core/ParserTest.php [hunk 2]
```diff
@@ -881,15 +881,15 @@ public function parserCorrectlyParsesComments01()
         $sourceCode = $this->readTypoScriptFixture('ParserTestTypoScriptComments01');
         $expected = array(); // Fixture contains only comments, so expect empty parse tree
         $actualParseTree = $this->parser->parse($sourceCode);
-        $this->assertEquals($expected, $actualParseTree, 'The parse tree was not as expected after parsing fixture `ParserTestTypoScriptComments01.ts2`');
+        $this->assertEquals($expected, $actualParseTree, 'The parse tree was not as expected after parsing fixture `ParserTestTypoScriptComments01..fusion`');
```
Comment (skurfuerst) [inline Neos.Fusion/Tests/Unit/Core/ParserTest.php]: comment: "..fusion"

File: Neos.Neos/Classes/Domain/Service/TypoScriptService.php [hunk 3]
```diff
@@ -45,31 +45,39 @@ class TypoScriptService
...
-    protected $legacySiteRootTypoScriptPattern = 'resource://%s/Private/TypoScripts/Library/Root.ts2';
+    protected $legacySiteRootTypoScriptPattern = 'resource://%s/Private/TypoScript/Root.ts2';
 
     /**
      * Pattern used for determining the TypoScript root file for autoIncludes
      *
      * @var string
      */
-    protected $autoIncludeTypoScriptPattern = 'resource://%s/Private/TypoScript/Root.ts2';
+    protected $autoIncludeTypoScriptPattern = 'resource://%s/Private/TypoScript/Root.fusion';
```
Comment (skurfuerst) [inline Neos.Neos/Classes/Domain/Service/TypoScriptService.php]: what about using "Private/Fusion" here as base path; adding an additional migration to move the path in all packages.

File: Neos.Neos/Classes/Domain/Service/TypoScriptService.php [hunk 4]
```diff
@@ -45,31 +45,39 @@ class TypoScriptService
      *
      * @var string
      */
-    protected $siteRootTypoScriptPattern = 'resource://%s/Private/TypoScript/Root.ts2';
+    protected $siteRootTypoScriptPattern = 'resource://%s/Private/TypoScript/Root.fusion';
```
Comment (skurfuerst) [inline Neos.Neos/Classes/Domain/Service/TypoScriptService.php]: see comment below

File: Neos.Neos/Classes/Domain/Service/TypoScriptService.php (hunk 3 above)
Comment (johannessteu) [inline Neos.Neos/Classes/Domain/Service/TypoScriptService.php]: So this PR should just provide the support for .fusion files and then go on with the next tasks

**Comment by johannessteu** [comment]
@skurfuerst thanks for your review! See my comments inline

**Pull request merged**